# **Proyecto II**
## Erwin Barriga

In [2]:
import pandas as pd

### CONJUNTO DE DATOS 1

#### **Earthquake dataset**

In [4]:
df1 = pd.read_csv("./Datos 1/earthquake_data.csv")

# Display all columns
pd.set_option('display.max_column', None)

print(df1.shape)
df1.head()

(782, 19)


title  magnitude         date_time  \
0  M 7.0 - 18 km SW of Malango, Solomon Islands        7.0  22-11-2022 02:03   
1      M 6.9 - 204 km SW of Bengkulu, Indonesia        6.9  18-11-2022 13:37   
2                                      M 7.0 -         7.0  12-11-2022 07:09   
3           M 7.3 - 205 km ESE of Neiafu, Tonga        7.3  11-11-2022 10:48   
4                                      M 6.6 -         6.6  09-11-2022 10:14   

   cdi  mmi  alert  tsunami  sig net  nst   dmin   gap magType    depth  \
0    8    7  green        1  768  us  117  0.509  17.0     mww   14.000   
1    4    4  green        0  735  us   99  2.229  34.0     mww   25.000   
2    3    3  green        1  755  us  147  3.125  18.0     mww  579.000   
3    5    5  green        1  833  us  149  1.865  21.0     mww   37.000   
4    0    2  green        1  670  us  131  4.998  27.0     mww  624.464   

   latitude  longitude                  location continent          country  
0   -9.7963    159.596  Malango, Solomon Islands   Oceania  Solomon Islands  
1   -4.9559    100.738       Bengkulu, Indonesia       NaN              NaN  
2  -20.0508   -178.346                       NaN   Oceania             Fiji  
3  -19.2918   -172.129             Neiafu, Tonga       NaN              NaN  
4  -25.5948    178.278                       NaN       NaN              NaN

#### 1) Fuente de datos

`https://www.kaggle.com/`

#### 2) Breve descripción de datos

`Dataset con registro de los terremotos ocurridos entre ene-2001 y ene-2023`

#### 3) ¿Cuál es el objetivo?

`Predecir si el evento sísmico genera un tsunami o no`

#### 4) ¿Este es un problema de clasificación o regresión?
`Clasificación`

#### 5) ¿Cuántas características hay?
`18` 

#### 6) ¿Cuántas filas de datos hay?

`782`

#### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

`El desbalanceo del evento a predecir`

In [5]:
df1["tsunami"].value_counts(normalize=True)

0    0.611253
1    0.388747
Name: tsunami, dtype: float64

### CONJUNTO DE DATOS 2

#### **Supermarket sales**

In [6]:
df2 = pd.read_csv("./Datos 2/supermarket_sales - Sheet1.csv")

# Display all columns
pd.set_option('display.max_column', None)

print(df2.shape)
df2.head()

(1000, 17)


Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   
4  373-73-7910      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   
4       Sports and travel       86.31         7  30.2085  634.3785   2/8/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4  
4  10:37      Ewallet  604.17                 4.761905       30.2085     5.3

#### 1) Fuente de datos

`https://www.kaggle.com/`

#### 2) Breve descripción de datos

`Dataset de ventas históricas de 3 supermercados diferentes`

#### 3) ¿Cuál es el objetivo?

`Predecir los ingresos brutos por la venta de productos`

#### 4) ¿Este es un problema de clasificación o regresión?
`Regresión`

#### 5) ¿Cuántas características hay?
`16`

#### 6) ¿Cuántas filas de datos hay?

`1000`

#### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

`Determinar las características significativas que ayuden a la predicción`